packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -45,212 +42,124 @@ import {
-    setShowModal((prev) => !prev);
-  };
-
-  const handleConfirmDelete = () => {
-    if (webhookToDelete) {
-      handleConfirmDeleteOne();
-    } else {
-      handleConfirmDeleteAll();
+    if (webhooks) {
+      notifyStatus(
+        formatMessage({
+          id: 'Settings.webhooks.list.loading.success',
+          defaultMessage: 'Webhooks have been loaded',
+        })
+      );
     }
-  };
+  }, [webhooks, webhooksError, toggleNotification, formatMessage, notifyStatus, formatAPIError]);
 
-  const handleConfirmDeleteOne = async () => {
-    try {
-      await del(`/admin/webhooks/${webhookToDelete}`);
-
-      dispatch({
-        type: 'WEBHOOK_DELETED',
-        index: getWebhookIndex(webhookToDelete),
+  const deleteMutation = useMutation(
+    async () => {
+      await post('/admin/webhooks/batch-delete', {
+        ids: webhooksToDelete,
       });
-    } catch (err) {
-      /**
-       * TODO: especially this.
-       */
-      if (err.code !== 20) {
+      setWebhooksToDelete([]);
+
+      setShowModal(false);
+    },
+    {
+      onError(error) {
         toggleNotification({
           type: 'warning',
-          message: { id: 'notification.error' },
+          message: formatAPIError(error),
         });
-      }
+      },
+      onSuccess() {
+        refetchWebhooks();
+      },
     }
-    setShowModal(false);
-  };
-
-  const handleConfirmDeleteAll = async () => {
-    const body = {
-      ids: webhooksToDelete,
-    };
+  );
 
-    try {
-      await post('/admin/webhooks/batch-delete', body);
+  const enabledMutation = useMutation(
+    async ({ isEnabled, id }) => {
+      const body = {
+        ...webhooks.find((webhook) => webhook.id === id),
+        isEnabled,
+      };
+      delete body.id;
 
-      if (isMounted.current) {
-        dispatch({
-          type: 'WEBHOOKS_DELETED',
+      await put(`/admin/webhooks/${id}`, body);
+    },
+    {
+      onError(error) {
+        toggleNotification({
+          type: 'warning',
+          message: formatAPIError(error),
         });
-      }
-    } catch (err) {
-      if (isMounted.current) {
-        if (err.code !== 20) {
-          toggleNotification({
-            type: 'warning',
-            message: { id: 'notification.error' },
-          });
-        }
-      }
-    }
-    setShowModal(false);
-  };
-
-  const handleDeleteClick = (id) => {
-    setShowModal(true);
-
-    if (id !== 'all') {
-      dispatch({
-        type: 'SET_WEBHOOK_TO_DELETE',
-        id,
-      });
+      },
+      onSuccess() {
+        refetchWebhooks();
+      },
     }
-  };
-
-  const handleEnabledChange = async (value, id) => {
-    const webhookIndex = getWebhookIndex(id);
-    const initialWebhookProps = webhooks[webhookIndex];
-    const keys = [webhookIndex, 'isEnabled'];
-
-    const body = {
-      ...initialWebhookProps,
-      isEnabled: value,
-    };
+  );
 
-    delete body.id;
+  const confirmDelete = async () => deleteMutation.mutateAsync();
```

Comment:
```suggestion
  const confirmDelete = async () => deleteMutation.mutate();
```

mutateAsync throws etc.